addons/account/tests/test_reconciliation.py

```diff
@@ -776,6 +776,43 @@ def test_unreconcile_exchange(self):
         credit_aml.with_context(invoice_id=inv.id).remove_move_reconcile()
         self.assertAlmostEquals(inv.residual, 111)
 
+    def test_revert_invoice_move(self):
+        company = self.env.ref('base.main_company')
+        company.tax_cash_basis_journal_id = self.cash_basis_journal
+        company.tax_exibility = True
+        in_invoice = self.create_invoice(
+            type='in_invoice', invoice_amount=50,
+            currency_id=self.currency_usd_id)
+        in_invoice.journal_id.update_posted = True
+        in_invoice.action_cancel()
+        in_invoice.state = 'draft'
+        in_invoice.invoice_line_ids.write({
+            'invoice_line_tax_ids': [(6, 0, [self.tax_cash_basis.id])]})
+        in_invoice.compute_taxes()
+        in_invoice.action_invoice_open()
+        reversed_move_list = in_invoice.move_id.reverse_moves()
+        self.assertEqual(len(reversed_move_list), 1)
+
+        tax_cash_basis_sale = self.tax_cash_basis.copy({
```

Comment:
split in_invoice test and out_invoice test using 2 test methods, please

Reply:
done